core_backend/app/user_tools/routers.py

```diff
@@ -7,15 +7,53 @@
 
 from ..auth.dependencies import get_current_user
 from ..database import get_async_session
-from ..users.models import UserDB, update_user_api_key
+from ..users.models import (
+    UserAlreadyExistsError,
+    UserDB,
+    save_user_to_db,
+    update_user_api_key,
+)
+from ..users.schemas import UserCreateWithPassword, UserRetrieve
 from ..utils import generate_key, setup_logger
 from .schemas import KeyResponse
 
-router = APIRouter(prefix="/key", tags=["API Key Management"])
+router = APIRouter(prefix="/user", tags=["User Tools"])
 logger = setup_logger()
 
 
-@router.put("/", response_model=KeyResponse)
+@router.post("/", response_model=UserRetrieve)
+async def create_user(
+    user: UserCreateWithPassword,
+    user_db: Annotated[UserDB, Depends(get_current_user)],
+    asession: AsyncSession = Depends(get_async_session),
+) -> UserRetrieve | None:
+    """
+    Create user endpoint. Can only be used by user with ID 1.
+    """
+
+    if user_db.user_id != 1:
```

Comment:
So if for reason `admin` ends up not having a `user_id` of 1, it will fail. I guess that's an ok corner case that we can look out for.

E.g. when we deploy this and run this script, `admin` will not have `user_id` of 1 since the db already has a few users.

Reply:
Basically there's no concept of admin haha it's just whoever was the first user is admin. We can be mindful of that until we implement proper superuser permission handling. Actually, we can probably just have an extra column in the user table that is "admin" which could be a bool, and we check for that instead of user_id. Will think about this more later!